src/benchttp/common.ts

```diff
@@ -1 +1,22 @@
 export type GoDuration = `${number}${'ns' | 'µs' | 'ms' | 's' | 'm' | 'h'}`
+
+export interface Statistics {
+  min: number
+  max: number
+  mean: number
+  stdDev: number
+  median: number
+  deciles: number[] | null
+  quartiles: number[] | null
```

Comment:
Suggestion: they could be a tuple of 10 and a tuple of 4 respectively.

```suggestion
  deciles: Tuple<number, 10> | null
  quartiles: Tuple<number, 4> | null
```

`Tuple` type must be defined however.

Although it may ask too much work for not enough benefits.

Reply:
Oh nice, didn't know that kind of stuff was even doable.
A bit unsatisfied with naming `Tuple`, as it usually allows mixed (but ordered) types, to me it looks more like a fixed length array: any suggestions? `FixedArray`, `BoundedArray`?

Reply:
`FixedArray` seems the way to go. 

Ideally we would remove length mutation method from the `Array` type (aka `'splice' | 'push' | 'pop' | 'shift' |  'unshift'`) and enforce no mutation of the `length` property.


In pseudo code:
```ts
type FixedArray<T, N> = Array<T> & Exclude<Array, MutationMethods> & { readonly length: N}
```

I guess building on that can give us something that works.

Reply:
Nope, that would be to easy 🙃 It won't work because arrays have length of type `number`, hence it produces an error when assigning them to `{ length: N }` because `number` can't be narrowed to `N`:
```ts
type FixedArray<T, N extends number> = Omit<T[], ArrayMutationMethods> & {
  readonly length: N
}
export const a: FixedArray2<string, 2> = ['a', 'b']
/*
Type 'string[]' is not assignable to type 'FixedArray2<string, 2>'.
  Type 'string[]' is not assignable to type '{ readonly length: 2; }'.
    Types of property 'length' are incompatible.
      Type 'number' is not assignable to type '2'.
*/
```

There a trick for that but it's a bit hacky:
```ts
export type FixedArray<
  T,
  N extends number,
  A extends ImmutableArray<T> = []
> = A extends {
  length: N
}
  ? A
  : FixedArray<T, N, [...A, T]>
```
It uses a recursive type, that increments the length of an empty array each iteration until the length matches `N`.

I don't bother the hacky part of it, but what is most disappointing here is that `ImmutableArray<T>` is ignored (mutable methods are still accessible, despite `ImmutableArray<T>` alone being correct), and I can't figure out why right now.

<details>
<summary><code>ImmutableArray</code></summary>
<pre lang="ts">
type ImmutableArray&lt;T> = readonly T[]
</pre>
</details>

As an alternative, the following fully works:
```ts
type FixedArray<T, N extends number> = ImmutableArray<T> & { readonly length: N }

export const a: FixedArray2<string, 2> = ['a', 'b'] as const // as const is mandatory
```
But I'm not a fan of the added constraint on _how_ the type should be used (it won't work unless used with `as const`).

Reply:
Solution in 1c7353c9770b0c63231dd99ae441c746a5a933ed`feat: TypedArray`.
It is based on the recursive solution, with inlined `readonly T[]`, `readonly [...A, T]` instead of a defined type `ImmutableArray<T>`.